mathics/builtin/system.py

```diff
@@ -633,24 +633,36 @@ class SetEnvironment(Builtin):
     summary_text = "set system environment variable(s)"
 
     def eval(self, rule, evaluation):
-        "SetEnvironment[rule_]"
+        "SetEnvironment[rule_Rule]"
         env_var_name, env_var_value = rule.elements
+        # WMA does not give an error message if env_var_name is not a String - weird.
+        if not isinstance(env_var_name, String):
+            return None
+
         if not (env_var_value is SymbolNone or isinstance(env_var_value, String)):
             evaluation.message("SetEnvironment", "value", env_var_value)
             return SymbolFailed
 
-        if isinstance(env_var_name, String):
-            # WMA does not give an error message if env_var_name is not a String - weird.
-            os.environ[env_var_name.value] = (
-                None if None is SymbolNone else env_var_value.value
-            )
+        os.environ[env_var_name.value] = (
+            None if None is SymbolNone else env_var_value.value
+        )
         return SymbolNull
 
     def eval_list(self, rules: Expression, evaluation: Evaluation):
         "SetEnvironment[{rules__}]"
+
+        # All the rules must be of the form
+        for rule in rules.elements:
+            if not rule.has_form("System`Rule", 2):
+                return None
+            if not isinstance(rule.elements[0], String):
+                return None
+
         for rule in rules.elements:
             self.eval(rule, evaluation)
-        return None
+        return SymbolNull
```

Comment:
If this is intended to be merged, then I suppose we should remove the duplicate return. 

If it is intended to be merged. LGTM.

Reply:
which one? I can merge the two if' s in lines 656 and 658, but I thought it was clearer in this way.

Reply:
I couldn't figure out how to do this with the +/- marker so I did this as PR https://github.com/Mathics3/mathics-core/pull/1430